src/const-eval.md

```diff
@@ -20,19 +20,25 @@ Additionally constant evaluation can be used to reduce the workload or binary
 size at runtime by precomputing complex operations at compiletime and only
 storing the result.
 
-Constant evaluation can be done by calling the `const_eval` query of `TyCtxt`.
+Constant evaluation can be done by calling the `const_eval_*` queries of `TyCtxt`.
 
-The `const_eval` query takes a [`ParamEnv`](./param_env.html) of environment in
-which the constant is evaluated (e.g. the function within which the constant is
-used) and a `GlobalId`. The `GlobalId` is made up of an
-`Instance` referring to a constant or static or of an
-`Instance` of a function and an index into the function's `Promoted` table.
+The `const_eval_*` queries use a [`ParamEnv`](./param_env.html) of environment
+in which the constant is evaluated (e.g. the function within which the constant is used)
+and a [`GlobalId`]. The `GlobalId` is made up of an `Instance` referring to a constant
+or static or of an `Instance` of a function and an index into the function's `Promoted` table.
 
-Constant evaluation returns a `Result` with either the error, or the simplest
+Constant evaluation returns a [`ConstEvalResult`] with either the error, or the simplest
 representation of the constant. "simplest" meaning if it is representable as an
-integer or fat pointer, it will directly yield the value (via `ConstValue::Scalar` or
-`ConstValue::ScalarPair`), instead of referring to the [`miri`](./miri.html) virtual
-memory allocation (via `ConstValue::ByRef`). This means that the `const_eval`
-function cannot be used to create miri-pointers to the evaluated constant or
+integer or fat pointer, it will directly yield the value (via [`ConstValue::Scalar`]
+or [`ConstValue::Slice`]), instead of referring to the [`miri`](./miri.html) virtual
+memory allocation (via [`ConstValue::ByRef`]). This means that the `const_eval_*`
+functions cannot be used to create miri-pointers to the evaluated constant or
 static. If you need that, you need to directly work with the functions in
-[src/librustc_mir/const_eval.rs](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir/const_eval/index.html).
+[`src/librustc_mir/const_eval.rs`].
```

Comment:
If you need the value of a constant inside Miri, the right function to call is `eval_const_to_op` (soon to be renamed, likely to `const_to_op` or so).

"miri-pointer to evaluated constant" does not make a lot of sense as constants do not have an address.

Reply:
Thanks! Addressed via e378c9386f6f5fc9c82b9dafa8d21103e0966881.